modules/relation/src/main/RelationActor.scala

```diff
@@ -103,4 +156,9 @@ private[relation] final class RelationActor(
         if (ids.nonEmpty) bus.publish(SendTos(ids.toSet, message, userId), 'users)
       }
     }
+
+  private def notifyFollowersFriendInStudyStateChanged(userId: String, studyId: String, message: String) =
+    api fetchFollowers userId map (_ filter onlines.contains) foreach { ids =>
```

Comment:
`fetchFollowers` is quite expensive. But it looks like your code is not calling it to aggressively, so we should be fine here. It would be a problem if it was called every time a study move is played, but I don't see that it's the case.